src/main/java/com/google/sps/app/servlets/LogIn.java

```diff
@@ -0,0 +1,61 @@
+package com.google.sps.app.servlets;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.api.gax.rpc.InvalidArgumentException;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.gson.Gson;
+import com.google.sps.util.Validator;
+
+@WebServlet("/LogIn")
+public class LogIn extends HttpServlet {
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String username = request.getParameter("text-input-user-name");
```

Comment:
use trim() to remove any trailing or starting whitespaces.

Reply:
Good idea. Done